pkg/util/chunkinfologger/debugheader_test.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkinfologger
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/grafana/dskit/middleware"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChunkInfoLoggerMiddleware(t *testing.T) {
+	testCases := map[string]struct {
+		headers       http.Header
+		expectedOk    bool
+		expectedValue []string
```

Comment:
[nit] `expectedLoggingSeriesLabels` might be clearer

(similar thing applies to `recordedValue` below)

Reply:
resolving to keep one discussion for the 4 variables